Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m

```diff
@@ -25,81 +26,184 @@
 % permissions and limitations under the License.                        %
 %-----------------------------------------------------------------------%
 
-% Three variables: 1) passive stiffness 
-%                  2) active max torque 
-%                  3) hop timing
-% See buildHopperFromSolution() for more details.
-numVariables = 3; 
+% CHOOSE PROBLEM: 'one_hop' or 'three_hops'
+problem = 'one_hop';
+switch problem
+    
+    % Optimize the height of one hop. The single optimization variable 
+    % determines when to simultaneously release a loaded passive device and 
+    % activate the hopper vastus muscle (see CONSTRUCTCONTROLS below). This
+    % simple univariate problem uses FMINBND from the MATLAB Optimization 
+    % Toolbox. 
+    case 'one_hop'
+        lb = 0;
+        ub = 1;         
+        options = optimset('TolFun', 1e-2, ...
+                           'MaxIter', 10000, ...
+                           'Display', 'iter'); 
+        [x, f] = fminbnd(@(x) objective(x,problem),lb,ub,options);
+                     
+    % Optimize the height of a sequence of three hops. The optimization
+    % variables decide to switch the actuators on and off to load and 
+    % release the passive device with the activated muscle (see 
+    % CONSTRUCTCONTROLS below). This problem is subject to falling into
+    % undesired local minima (e.g. solutions with less than three hops) and
```

Comment:
In my view: the problem *has* local minima. Being subject to falling into the local minima is an attribute of the solver. Consider "We use a global optimizer (genetic algorithm from MATLAB's Global Optimization Toolbox) to avoid falling into local minima (e.g., solutions with less than three hops)."

Reply:
Good points. I'll incorporate this phrasing too.

Reply:
nitpicky: "less" --> "fewer"